sources/en/r/ranobes.py

```diff
@@ -1,109 +1,104 @@
 # -*- coding: utf-8 -*-
 import logging
+import re
 
-import js2py
-from bs4.element import Tag
+from typing import Generator, Union
 
-from lncrawl.core.crawler import Crawler
+from bs4 import BeautifulSoup, Tag
+
+from lncrawl.models import Chapter, SearchResult, Volume
+from lncrawl.templates.browser.searchable import SearchableBrowserTemplate
+from lncrawl.core.exeptions import FallbackToBrowser
+
+from urllib.parse import urljoin, quote_plus
 
 logger = logging.getLogger(__name__)
 
 
-class RanobeLibCrawler(Crawler):
+digit_regex = re.compile(r"\/(\d+)-")
+
+
+class RanobeLibCrawler(SearchableBrowserTemplate):
     base_url = [
-        "http://ranobes.net/",
-        "https://ranobes.net/",
+        "https://ranobes.top/",
     ]
```

Comment:
It only supports https?

Reply:
Also, don't want to ask much since I'm not the one doing the work, but would it be possible just to convert .net links to .top?

To maintain cross-compatibility with novels people could already have.